### PR TITLE
Filter by `organizationId` in `ListWorkspaces`

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

With this PR, you can optionally pass an `organization_id` to `ListWorkspaces` for workspace filtering akin to the Dashboard.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 38fc47f</samp>

Added support for listing workspaces by organization in the experimental v1 workspaces API. Extended the proto message, the TypeScript client library, the Go protocol service, and the public API server with a new `organization_id` field and parameter.

</details>

## Related Issue(s)

Fixes EXP-287

## How to test

Create two orgs and in each make a workspace. Then note down the organization ID and with a PAT execute this:
```bash
curl -X POST 'https://api.ft-papi-li91b56fd16f.preview.gitpod-dev.com/gitpod.experimental.v1.WorkspacesService/ListWorkspaces' \
-H 'Authorization: Bearer gitpod_pat_token' \
-H 'Content-Type: application/json' \
-d '{"organizationId":"org_id"}' | jq .
```

This should only give the one workspace back to you. 

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-papi-li91b56fd16f</li>
	<li><b>🔗 URL</b> - <a href="https://ft-papi-li91b56fd16f.preview.gitpod-dev.com/workspaces" target="_blank">ft-papi-li91b56fd16f.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-papi-listworkspaces-org-id-gha.14507</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/gitpod-protocol/go/gitpod-service.go [hunk 1]
```diff
@@ -1490,9 +1490,11 @@ type UserInfo struct {
 
 // GetWorkspacesOptions is the GetWorkspacesOptions message type
 type GetWorkspacesOptions struct {
```
[the remark this answers is not in the document]
Comment (akosyakov) [reply components/gitpod-protocol/go/gitpod-service.go:1492]: We don't need to be consistent with Server API. We design a new API.

File: components/gitpod-protocol/go/gitpod-service.go [hunk 2]
```diff
@@ -1490,9 +1490,11 @@ type UserInfo struct {
...
-	Limit        float64 `json:"limit,omitempty"`
-	PinnedOnly   bool    `json:"pinnedOnly,omitempty"`
-	SearchString string  `json:"searchString,omitempty"`
+	Limit                 float64 `json:"limit,omitempty"`
+	SearchString          string  `json:"searchString,omitempty"`
+	PinnedOnly            bool    `json:"pinnedOnly,omitempty"`
+	IncludeWithoutProject bool    `json:"includeWithoutProject,omitempty"`
```
Comment (akosyakov) [inline components/gitpod-protocol/go/gitpod-service.go:1496]: includeWithoutProject is it neccessary?

**Review by akosyakov — APPROVED** [review]
generally looks good, please don't just copy things from server API

/hold

File: components/gitpod-protocol/go/gitpod-service.go (hunk 2 above)
Comment (filiptronicek) [reply components/gitpod-protocol/go/gitpod-service.go:1496]: @akosyakov it isn't, but it should be the same as in our [typescript source](https://github.com/gitpod-io/gitpod/blob/0d22d834b1c063f59db3d6af8f90460db9e128ff/components/gitpod-protocol/src/gitpod-service.ts#L382). This is just trying to reflect that, not adding anything to public API.

Comment (akosyakov) [reply components/gitpod-protocol/go/gitpod-service.go:1496]: again, typescript is not a source, it is completely another API

**Pull request merged**